web/app/scripts/navbar/navbar-controller.js

```diff
@@ -0,0 +1,60 @@
+(function () {
+    'use strict';
+
+    /* ngInject */
+    function NavbarController($log, $rootScope, $state, $stateParams, RecordTypes) {
+        var ctl = this;
+        ctl.updateState = updateState;
+
+        setRecordTypes();
+        setStates();
+        $rootScope.$on('$stateChangeSuccess', setStates);
+
+        // Sets the list of available record types, and also the active one if provided
+        function setRecordTypes() {
+            RecordTypes.query({ active: 'True' })
+                .$promise.then(function(recordTypes) {
+                    ctl.recordTypes = recordTypes;
+                    if (!recordTypes.length) {
+                        $log.warn('No record types returned');
+                    } else {
+                        if ($stateParams.rtuuid) {
+                            var index = _.findIndex(recordTypes, function(rt) {
+                                return rt.uuid === $stateParams.rtuuid;
+                            });
+
+                            if (index < 0) {
+                                $log.warn('Record type UUID is invalid');
+                            } else {
+                                ctl.selectedRecordType = recordTypes[index];
+                            }
+                        }
+
+                        if (!ctl.selectedRecordType) {
+                            ctl.selectedRecordType = recordTypes[0];
+                        }
+                    }
+                });
+        }
+
+        // Sets states that can be navigated to (exclude current state, since we're already there)
+        function setStates() {
+            ctl.selectedState = $state.current;
+            ctl.availableStates = _.chain(['home', 'dashboard', 'map', 'record.list','record.add'])
```

Comment:
Alright, I'll take a look. We definitely don't want to grab all the configured states, because we only want to show some specific ones. A good approach may be to add a variable to the state config for 'showInNavbar' and use that for filtering.